test/functional/notes_controller_test.rb

```diff
@@ -49,6 +49,85 @@ class NotesControllerTest < ActionController::TestCase
       end
     end
 
+    context "when an Important note is provided" do
+      should "show 'Note recorded' and save a completed note" do
+        post :create,
+             params: {
+               edition_id: @edition.id,
+               note: {
+                 type: "important_note",
+                 comment: @note_text,
+               },
+             }
+
+        @edition.reload
+
+        assert_equal(@note_text, @edition.important_note.comment)
+        assert_redirected_to history_edition_path(@edition)
+        assert_includes flash[:success], "Note recorded"
+      end
+
+      should "show 'Note resolved' if a parent note exists and then an empty note is saved" do
+        post :create,
+             params: {
+               edition_id: @edition.id,
+               note: {
+                 type: "important_note",
+                 comment: "Parent note",
+               },
+             }
+
+        @edition.reload
+
+        post :create,
+             params: {
+               edition_id: @edition.id,
+               note: {
+                 type: "important_note",
+                 comment: "",
+               },
+             }
+        @edition.reload
+
+        assert_nil(@edition.important_note)
+        assert_redirected_to history_edition_path(@edition)
+        assert_includes flash[:success], "Note resolved"
+      end
+
+      should "show no flash message if no parent note exists and then an empty note is saved" do
+        post :create,
+             params: {
+               edition_id: @edition.id,
+               note: {
+                 type: "important_note",
+                 comment: "",
+               },
+             }
+
+        assert_nil(@edition.important_note)
+        assert_redirected_to history_edition_path(@edition)
+        assert flash.empty?, "Expected no flash message, but found: #{flash.inspect}"
+      end
+
+      should "show 'Note failed to save' if an error occurs while saving the Important note" do
+        @user.stubs(:record_note).with(@edition, @note_text, "important_note").returns(false)
+
+        post :create,
+             params: {
+               edition_id: @edition.id,
+               note: {
+                 type: "important_note",
+                 comment: @note_text,
+               },
+             }
+
+        @edition.reload
+
+        assert_redirected_to history_update_important_note_edition_path(@edition)
+        assert_includes flash[:danger], "Note failed to save"
```

Comment:
Would `assert_equal` be more appropriate here?
```suggestion
        assert_equal "Note failed to save", flash[:danger]
```